apis/common/v1/resource.go

```diff
@@ -136,6 +136,14 @@ type ResourceSpec struct {
 	// +optional
 	WriteConnectionSecretToReference *SecretReference `json:"writeConnectionSecretToRef,omitempty"`
 
+	// PublishConnectionDetailsTo specifies the connection secret config which
+	// contains a name, metadata and a reference to secret store config to
+	// which any connection details for this managed resource should be written.
+	// Connection details frequently include the endpoint, username,
+	// and password required to connect to the managed resource.
+	// +optional
+	PublishConnectionDetailsTo *PublishConnectionDetailsTo `json:"publishConnectionDetailsTo,omitempty"`
```

Comment:
Cross reference to https://github.com/crossplane/crossplane/issues/2929 - might be worth considering before we introduce this new field.

Reply:
Responded there: https://github.com/crossplane/crossplane/issues/2929#issuecomment-1060493213

Reply:
Either way, it's probably worth documenting here and/or on WriteConnectionSecretToRef what happens when both are set / what their relationship is.